MusicHub/tracks/views.py

```diff
@@ -0,0 +1,91 @@
+from django.utils.decorators import method_decorator
+
+from drf_yasg.utils import swagger_auto_schema
+
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from MusicHub.main.utils import LargeResultsSetPagination
+from MusicHub.tracks import custom_track_schema
+from MusicHub.tracks.models import Track
+from MusicHub.tracks.serializers import (
+    CreateTrackSerializer,
+    ListTrackSerializer,
+)
+
+
+class UploadTrackView(generics.CreateAPIView):
+    """
+    View for uploading new track by user
+    """
+
+    permission_classes = [IsAuthenticated]
+    serializer_class = CreateTrackSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = CreateTrackSerializer(
+            data=request.data, context={"user": request.user}
+        )
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(status=201, data=serializer.data)
```

Comment:
again, why we need to reimplement DRF methods https://www.django-rest-framework.org/tutorial/3-class-based-views/#using-generic-class-based-views ?